crates/ark-contracts/arkchain/src/orderbook.cairo

```diff
@@ -216,22 +226,20 @@ mod orderbook {
                     }
                 );
         }
-        fn _create_listing_offer(
-            ref self: ContractState, order: OrderV1, order_type: OrderType, order_hash: felt252
-        ) {
-            self
-                .emit(
-                    OrderPlaced {
-                        order_hash: order_hash,
-                        order_version: order.get_version(),
-                        order_type: order_type,
-                        order: order,
-                    }
-                );
-        }
-        fn _create_listing_collection_offer(
+
+        fn _create_offer(
             ref self: ContractState, order: OrderV1, order_type: OrderType, order_hash: felt252
         ) {
+            // Manage auction offer
+            let ressource_hash = order.compute_ressource_hash();
+            let (auction_order_hash, auction_end_date) = self.auctions.read(ressource_hash);
+            if auction_order_hash.is_non_zero() {
+                let auction_order = self.auction_offers.read(order_hash);
+                assert(auction_order.is_zero(), 'already existing auction offer');
+                self.auction_offers.write(order_hash, auction_order_hash);
+            }
```

Comment:
@kwiss, could you take a close look at this code to see if I've correctly understood how to use the storage?